packages/sdk-btc/source/transaction.service.ts

```diff
@@ -1,55 +1,221 @@
+			if (input.signatory.actsWithMnemonic()) {
+				console.log(input.signatory.signingKey());
 			}
+			const network = getNetworkConfig(this.configRepository);
 
-			// NOTE: this is a WIF/PrivateKey - should probably be passed in as wif instead of mnemonic
+			// 1. Derive the sender address (corresponding to first address index for the wallet)
+			const { address, type, path } = await this.addressService.fromMnemonic(
+				input.signatory.signingKey(),
+				identityOptions,
+			);
+			console.log(type, address, path);
 
-			// 1. Derive the sender address
-			const { address } = await this.addressService.fromWIF(input.signatory.signingKey());
-			// ({ wif: input.signatory.signingKey() });
+			// 3. Compute the amount to be transferred
+			const amount = this.toSatoshi(input.data.amount).toNumber();
 
-			// 2. Aggregate the unspent transactions
-			const unspent: UnspentTransaction[] = await this.#unspent.aggregate(address);
+			// 4. Add utxos
+			const accountKey = BIP32.fromMnemonic(input.signatory.signingKey(), network)
+				.deriveHardened(levels.purpose!)
+				.deriveHardened(levels.coinType)
+				.deriveHardened(levels.account || 0);
 
-			// 3. Compute the amount to be transfered
-			const amount = this.toSatoshi(input.data.amount).toNumber();
+			console.log("accountKey", accountKey.toBase58(), accountKey.neutered().toBase58());
 
-			// 4. Build and sign the transaction
-			let transaction = new Transaction().from(unspent).to(input.data.to, amount).change(address);
+			const targets = [
+				{
+					address: input.data.to,
+					value: amount,
+				},
+			];
 
-			// 5. Set a fee if configured. If none is set the fee will be estimated by bitcore-lib.
-			if (input.fee) {
-				const fee = this.toSatoshi(input.fee).toNumber();
-				transaction = transaction.fee(fee);
-			}
+			const psbt = new bitcoin.Psbt({ network: network });
+
+			await this.#addUtxos(psbt, levels, accountKey, targets, address);
 
-			return transaction.sign(input.signatory.signingKey()).toString();
+			const transaction: bitcoin.Transaction = psbt.extractTransaction();
+
+			return this.dataTransferObjectService.signedTransaction(
+				transaction.getId(),
+				{
+					sender: address,
+					recipient: input.data.to,
+					amount: amount,
+					fee: psbt.getFee(),
+					timestamp: new Date(), // TODO See if we have a proper timestamp inside the built transaction
+				},
+				transaction.toHex(),
+			);
 		} catch (error) {
+			console.log(error);
 			throw new Exceptions.CryptoException(error);
 		}
 	}
 
-	@IoC.postConstruct()
-	private onPostConstruct(): void {
-		this.#unspent = new UnspentAggregator({
-			http: this.httpClient,
-			peer: Helpers.randomHostFromConfig(this.configRepository),
+	async #addUtxos(psbt, levels: Levels, accountKey: BIP32Interface, targets, changeAddress): Promise<void> {
+		const feeRate = 55; // satoshis per byte // @TODO Need to get this from endpoint
```

Comment:
This should be retrieved through the `FeeService` if this is needed to calculate the transaction fee.